platform/security/common/src/test/java/org/codice/ddf/security/common/SPMetadataParserTest.java

```diff
@@ -0,0 +1,107 @@
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.security.common;
+
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+import com.google.common.collect.ImmutableSet;
+import ddf.security.samlp.SamlProtocol.Binding;
+import ddf.security.samlp.impl.EntityInformation;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class SPMetadataParserTest {
+  private static List<String> spMetadata;
+
+  private static Set<Binding> bindingSet;
+
+  private static final Map EMPTY_MAP = Collections.emptyMap();
+
+  private static final String FILE_PATH = "src/test/resources/SPMetadata";
+
+  private static final String ENTITY_ID = "https://localhost:8993/services/saml";
+  private static final String LOGOUT_URL = "https://localhost:8993/services/saml/logout";
+  private static final String ACS_URL = "https://localhost:8993/services/saml/sso";
+  private static final String CERTIFICATE =
+      "MIICsjCCAhugAwIBAgIGAWBQwru6MA0GCSqGSIb3DQEBCwUAMIGEMQswCQYDVQQGEwJVUzELMAkGA1UECBMCQVoxDDAKBgNVBAoTA0RERjEMMAoGA1UECxMDRGV2MRkwFwYDVQQDExBEREYgRGVtbyBSb290IENBMTEwLwYJKoZIhvcNAQkBFiJlbWFpbEFkZHJlc3M9ZGRmcm9vdGNhQGV4YW1wbGUub3JnMCAXDTE3MTIxMjE2NDM0N1oYDzIxMTcxMjEyMTY0MzQ3WjAUMRIwEAYDVQQDDAlsb2NhbGhvc3QwggEiMA0GCSqGSIb3DQEBAQUAA4IBDwAwggEKAoIBAQDEwOqpKuMr+GmiT3kKcMKp4361ByVBaxvj5M6rptn+Al8oOZfxfaDOhOikUVuaPTuIYvTWW04KaQNHI3nm86RwldbwMlPZka5jGeYkOJ3qah1C6OMmMDEC7lM0/bhxPLc8C0cUwAG8FdlBLDKCYFjHj6UBq5xhD/1uJyvSoWcFqJQbXP+igVV7tDdRxRDhp8f5eActecjQXK5+bpzNs8KC+3gvnhBj/ux/DYyrE6iA4nKMnelIaZjhW11bsuQwM6XK8Tq7PQ2LuAuhFoKUq/p8p8Sva75/3nGeCHOlcjj7rnoSCcUgX+u1nnCN/wsshyRAX30wS7C75nQilQkxCfAlAgMBAAGjGDAWMBQGA1UdEQQNMAuCCWxvY2FsaG9zdDANBgkqhkiG9w0BAQsFAAOBgQAQBBked2MsqlmIwxSUXLee6f9R14GPsNsOkLohFhBTKuxC++n+q0+LvMRLmFxPj4dois/dHKQiD7wYWYY4WPvb3nD2aPev4NZm8erB9+XPVueRzxaLXQ+K1oW9Kjui8+2q1lTBBGOjma1DT/b6FVhyJo5SjQ3JMKe9TACVp/YA7A==";
+
+  @BeforeClass
+  public static void setupClass() throws Exception {
+
+    // read SPMetadata file into spMetadata
+    spMetadata = new ArrayList<>();
+    try (Stream<String> lines = Files.lines(Paths.get(FILE_PATH))) {
```

Comment:
✏️ It would be good to confirm that we can parse a set of multiple sp metadata

Reply:
I don't know if this is planned functionality or not, but the only the last metadata's map is returned. Adding extra different xmls will only provide the latest one's map, so rather than complicating the code with more xml files I just copied the first index of `SPMetadata` into the second index to test that it won't break with multiple entries.

Reply:
It looks like no matter how many metadata sources you give `SPMetadataParser.parse` it always returns a map with one entry, that entry being the last string in the `spMetadata` list. I'm going to assume this is the wanted functionality.

Rather than mucking up the tests with more xmls I just opted to copy the xml to the second index in `spMetadata` as well just to make sure multiple entries won't break.

Reply:
So if you give it a list of metadatas it will only parse and return one? I don't think that's the right. Why would we bother providing a list then?

Reply:
I agree with you. Maybe that is what it is supposed to do but it seems really unlikely and really jankey. That's also a question of whether or not if falls under the jurisdiction of my ticket, or if we would need to make another ticket.

Reply:
It's probably because you are giving it the same SP twice. The key is the entityID so it's probably replacing the first results with the second. Try giving it a different one. I can give you Spring's metadata if you would like.